### PR TITLE
fix: handles invalid export api result amplitude

### PR DESCRIPTION
**Pull request by a-rampalli**
## What
https://www.notion.so/rudderstacks/Amplitude-events-resource-error-9e42e4eae7934807a8c79064e676b910

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- [ ] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [ ] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
- [ ] After the connector is published, connector added to connector index as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
- [ ] Seed specs have been re-generated by building the platform and committing the changes to the seed spec files, as described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

<details><summary><strong>Connector Generator</strong></summary>

- [ ] Issue acceptance criteria met
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)
- [ ] If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- [ ] The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- [ ] Documentation which references the generator is updated as needed

</details>

## Tests

<details><summary><strong>Unit</strong></summary>

*Put your unit tests output here.*

</details>

<details><summary><strong>Integration</strong></summary>

*Put your integration tests output here.*

</details>

<details><summary><strong>Acceptance</strong></summary>

*Put your acceptance tests output here.*

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py [hunk 1]
```diff
@@ -176,12 +176,15 @@ def parse_response(self, response: requests.Response, stream_state: Mapping[str,
...
 
         for gzip_filename in zip_file.namelist():
-            with zip_file.open(gzip_filename) as file:
-                for record in self._parse_zip_file(file):
-                    if record[self.cursor_field] >= state_value:
-                        user_id_key = f"{self.name}_user_id" # to avoid name collision with user_id field
-                        record[user_id_key] = record.pop("user_id")
-                        yield self._date_time_to_rfc3339(record)  # transform all `date-time` to RFC3339
+            try:
+                with zip_file.open(gzip_filename) as file:
```
Comment (nidhilashkari17) [inline airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:180]: @a-rampalli In the case where it is failing to open zip file, won't adding try/catch lead to data loss?

Comment (a-rampalli) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:180]: Yes. But based on my understanding this error is caused due to faulty response from API even though we get 200 in response. Any suggestions on how we can avoid data loss?

Comment (a-rampalli) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:180]: API response for https://amplitude.com/api/2/export\?start\=20220907T20\&end\=20220908T19

Screenshot 2023-02-08 at 2 47 34 PM

Comment (debanjan97) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py:180]: This API response doesn't indicate a failure, AFAIR this API returns a zip file, which is supposed to be non-readable.

Comment (a-rampalli) [reply airbyte-integrations/connectors/source-amplitude/source_amplitude/api.py]: Yes. Turns out actual issue is caused due to not handling null date-time values in api response. Fixed it now.

**Review by debanjan97 — COMMENTED** [review]
Adding a commit ourselves will result in a merge conflict.
I think we can cherry-pick commits from the upstream branch (I could be wrong) and merge this PR with a merge commit instead of the usual squashing to avoid merge conflicts in future.
If cherry-picking is feasible, let's do that.

**Comment by a-rampalli** [comment]
Raised different PR for this which cherry picks required commit directly from airbyte upstream

**Pull request closed without merging**